cmd/plugin/main.go

```diff
@@ -0,0 +1,386 @@
+		Short: "A kubectl plugin for inspecting your ingress-nginx deployments",
+	}
+
+	// Respect some basic kubectl flags like --namespace
+	flags := genericclioptions.NewConfigFlags()
+	flags.AddFlags(rootCmd.PersistentFlags())
+
+	ingCmd := &cobra.Command{
+		Use:     "ingresses",
+		Aliases: []string{"ingress", "ing"},
+		Short:   "Provide a short summary of all of the ingress definitions",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			host, err := cmd.Flags().GetString("host")
+			if err != nil {
+				return err
+			}
+
+			allNamespaces, err := cmd.Flags().GetBool("all-namespaces")
+			if err != nil {
+				return err
+			}
+
+			util.PrintError(ingresses(flags, host, allNamespaces))
+			return nil
+		},
+	}
+	ingCmd.Flags().String("host", "", "Show just the ingress definitions for this hostname")
+	ingCmd.Flags().Bool("all-namespaces", false, "Find ingress definitions from all namespaces")
+	rootCmd.AddCommand(ingCmd)
+
+	confCmd := &cobra.Command{
+		Use:   "conf",
+		Short: "Inspect the generated nginx.conf",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			host, err := cmd.Flags().GetString("host")
+			if err != nil {
+				return err
+			}
+
+			pod, err := cmd.Flags().GetString("pod")
+			if err != nil {
+				return err
+			}
+
+			util.PrintError(conf(flags, host, pod))
+			return nil
+		},
+	}
+	confCmd.Flags().String("host", "", "Print just the server block with this hostname")
+	confCmd.Flags().String("pod", "", "Query a particular ingress-nginx pod")
+	rootCmd.AddCommand(confCmd)
+
+	generalCmd := &cobra.Command{
+		Use:   "general",
+		Short: "Inspect the other dynamic ingress-nginx information",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			pod, err := cmd.Flags().GetString("pod")
+			if err != nil {
+				return err
+			}
+
+			util.PrintError(general(flags, pod))
+			return nil
+		},
+	}
+	generalCmd.Flags().String("pod", "", "Query a particular ingress-nginx pod")
+	rootCmd.AddCommand(generalCmd)
+
+	infoCmd := &cobra.Command{
+		Use:   "info",
+		Short: "Show information about the ingress-nginx service",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			util.PrintError(info(flags))
+			return nil
+		},
+	}
+	rootCmd.AddCommand(infoCmd)
+
+	backendsCmd := &cobra.Command{
+		Use:   "backends",
+		Short: "Inspect the dynamic backend information of an ingress-nginx instance",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			pod, err := cmd.Flags().GetString("pod")
+			if err != nil {
+				return err
+			}
+			backend, err := cmd.Flags().GetString("backend")
+			if err != nil {
+				return err
+			}
+			onlyList, err := cmd.Flags().GetBool("list")
+			if err != nil {
+				return err
+			}
+			if onlyList && backend != "" {
+				return fmt.Errorf("--list and --backend cannot both be specified")
+			}
+
+			util.PrintError(backends(flags, pod, backend, onlyList))
+			return nil
+		},
+	}
+	backendsCmd.Flags().String("pod", "", "Query a particular ingress-nginx pod")
```

Comment:
What happens when we don't specify this?

Reply:
oh nevermind, https://github.com/kubernetes/ingress-nginx/pull/3779/files#diff-99cbd32a1c30f61ec2f0f0b744cf1f35R55